src/main/java/com/google/sps/data/Keywords.java

```diff
@@ -0,0 +1,105 @@
+package com.google.sps.data;
+
+import java.io.IOException;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
+import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
+import com.google.cloud.language.v1.EncodingType;
+import com.google.cloud.language.v1.Entity;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Provide methods for the storage, extraction, and retrieval of keywords.
+ */
+public final class Keywords {
+
+  private static final int MAX_NUM_KEYWORDS = 10;
+  public static final String KEY = "keywords";
+
+  /**
+   * @return a list of the 10 most salient keywords
+   */
+  public static Set<String> getKeywords() {
+    Query query = new Query(KEY).addSort("salience", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    HashSet<String> keywordSet = new HashSet<>();
+    int index = 0;
+    for (com.google.appengine.api.datastore.Entity entity : results.asIterable()) {
+      keywordSet.add((String) entity.getProperty("keyword"));
+
+      index++;
+      if (index >= MAX_NUM_KEYWORDS) {
+        break;
+      }
+    }
+    return keywordSet;
+  }
+
+  /**
+   * Adds the labels from the analysis of an image into the datastore.
+   */
+  public static void addKeywords(List<EntityAnnotation> blobAnalysis) throws IOException {
+    String labels = "";
+    for (EntityAnnotation label : blobAnalysis) {
+      labels += label.getDescription() + ", ";
+    }
+    addToDatastore(labels);
+  }
+
+  /**
+   * Adds the salient keywords from the analysis of a textual user input into the datastore.
+   */
+  public static void addKeywords(String message) throws IOException {
+    addToDatastore(message);
```

Comment:
What's the point of using this function instead of calling addToDatastore?

Reply:
It streamlines this method with the other method that takes input from the ImageServlet to create a set of salient keywords. So instead of having to call addToDatastore() for text-inputs, and addKeywords() for the ImageServlet, it makes more sense to have both call the same method (but different parameters) to achieve the same purpose.